File: python-service/app/models/video_chapter.py
# ============================================================
# FILE 5: app/models/video_chapter.py
# ============================================================
"""
VideoChapter Model - AI-detected video chapters
"""
from sqlalchemy import Column, Integer, String, Text, ForeignKey, UniqueConstraint
from sqlalchemy.orm import relationship
from sqlalchemy.dialects.postgresql import ARRAY  # ✅ Changed from JSONB
from app.core.database import Base


class VideoChapter(Base):
    __tablename__ = "video_chapters"
    __table_args__ = (
        UniqueConstraint('video_lesson_id', 'chapter_number', name='uq_video_chapter'),
        {'schema': 'academic'}
    )

    id = Column(Integer, primary_key=True, index=True)

    # ==================== Relationship ====================
    video_lesson_id = Column(Integer, ForeignKey('academic.video_lessons.id'), nullable=False)
    
    video_lesson = relationship("VideoLesson", back_populates="chapters")

    # ==================== Chapter Details ====================
    chapter_number = Column(Integer, nullable=False)
    title = Column(String(200), nullable=False)
    start_time_seconds = Column(Integer, nullable=False)
    end_time_seconds = Column(Integer, nullable=False)

    # ==================== AI Analysis ====================
    key_concepts = Column(ARRAY(String))  # ✅ PostgreSQL text[] array type
    summary = Column(Text)

File: java-service/src/main/java/com/edu/platform/model/VideoChapter.java
package com.edu.platform.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.util.List;

@Entity
@Table(name = "video_chapters", schema = "academic",
       uniqueConstraints = @UniqueConstraint(columnNames = {"video_lesson_id", "chapter_number"}))
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class VideoChapter {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "video_lesson_id", nullable = false)
    private VideoLesson videoLesson;
    
    @Column(name = "chapter_number", nullable = false)
    private Integer chapterNumber;
    
    @Column(name = "title", nullable = false, length = 200)
    private String title;
    
    @Column(name = "start_time_seconds", nullable = false)
    private Integer startTimeSeconds;
    
    @Column(name = "end_time_seconds", nullable = false)
    private Integer endTimeSeconds;
    
    // ✅ Use List<String> instead of String[] - works better with Hibernate 6
    @JdbcTypeCode(SqlTypes.ARRAY)
    @Column(name = "key_concepts", columnDefinition = "text[]")
    private List<String> keyConcepts;
    
    @Column(name = "summary", columnDefinition = "TEXT")
    private String summary;
}